File: 그리디/큰수의법칙-하윤.js
const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

const question = (q = '') => new Promise( (resolve) => rl.question(q, resolve));

async function main() {
    let [n, m, k] = (await question()).split(' ').map( s => Number(s));
    let l = (await question()).split(' ').map( s => Number(s)).sort( (a,b) => b-a);

    let count = 0;
    let result = 0;
    while (m-- !== 0) {
        if (count !== k) {
            count++;
            result += l[0];
        } else {
            count = 0;
            result += l[1];
        }
    }
    console.log(result);
}
main().then();

File: 그리디/큰수의 법칙 - 하윤.swift
var q = readLine()!.split(separator: " ").map { Int($0)! }
var l = readLine()!.split(separator: " ").map { Int($0)! }.sorted(by: {$0 > $1});

func main(q: [Int], l: [Int]) -> Void {
    var m = q[1]
    var count: Int = 0
    var result: Int = 0
    
    while m != 0 {
        if (count != q[2]) {
            count += 1
            result += l[0]
        } else {
            count = 0
            result += l[1]
        }
        m -= 1
    }
    print(result)
}
main(q: q, l: l)
